gax-java/gax-httpjson/src/main/java/com/google/api/gax/httpjson/HttpJsonCallOptions.java

```diff
@@ -29,23 +29,39 @@
  */
 package com.google.api.gax.httpjson;
 
+import static com.google.api.gax.util.TimeConversionUtils.toJavaTimeDuration;
+import static com.google.api.gax.util.TimeConversionUtils.toJavaTimeInstant;
+import static com.google.api.gax.util.TimeConversionUtils.toThreetenDuration;
+import static com.google.api.gax.util.TimeConversionUtils.toThreetenInstant;
+
+import com.google.api.core.ObsoleteApi;
 import com.google.auth.Credentials;
 import com.google.auto.value.AutoValue;
 import com.google.protobuf.TypeRegistry;
-import java.time.Duration;
 import javax.annotation.Nullable;
-import org.threeten.bp.Instant;
 
 /** Options for an http-json call, including deadline and credentials. */
 @AutoValue
 public abstract class HttpJsonCallOptions {
   public static final HttpJsonCallOptions DEFAULT = newBuilder().build();
 
   @Nullable
-  public abstract Duration getTimeout();
+  @ObsoleteApi("Use getTimeoutDuration() instead")
+  public abstract org.threeten.bp.Duration getTimeout();
+
+  @Nullable
+  public java.time.Duration getTimeoutDuration() {
+    return toJavaTimeDuration(getTimeout());
+  }
 
   @Nullable
-  public abstract Instant getDeadline();
+  @ObsoleteApi("Use getDeadlineInstant() instead")
+  public abstract org.threeten.bp.Instant getDeadline();
+
+  @Nullable
+  public final java.time.Instant getDeadlineInstant() {
```

Comment:
Should this be final?

Reply:
The purpose is to restrict the behavior of this method to its `threeten` counterpart. If by any chance we want to extend this class, we will want the `java.time` methods to have the same behavior as the `threeten` ones. Maybe there is another case I'm missing where `final` brings problems?